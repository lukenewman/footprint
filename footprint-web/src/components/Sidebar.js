import React from 'react';
import { Layout, Menu } from 'antd';
import 'antd/dist/antd.css';

const { Sider } = Layout;

/* 
  The main navigational component of the app. Clicking on a category
    will switch the app's content to show the selected category's emission types

  Props:
    categories ([Category]): dynamically supplied by the API
    onCategoryClicked ((category) => ()): callback for navigational purposes
*/
class Sidebar extends React.Component {
  menuItemClicked = ({ key }) => {
    this.props.onCategoryClicked(this.props.categories[key]);
  }

  render() {
    return (
      <Sider width={'20%'}>
        <Menu
          mode="inline"
          defaultSelectedKeys={['0']}
          style={{ height: '100%', borderRight: 0 }}
          onClick={this.menuItemClicked}
        >
          {this.props.categories.map((category, index) => (
            <Menu.Item key={index}>{category.name}</Menu.Item>
          ))}
        </Menu>
      </Sider>
    );
  }
}

export default Sidebar;