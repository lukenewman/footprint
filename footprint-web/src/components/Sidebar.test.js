import React from "react";
import renderer from 'react-test-renderer';
import Sidebar from "./Sidebar.js";

test("<Sidebar /> matches snapshot", () => {
  let categories = [
    {
      id: 0,
      name: "Housing"
    },
    {
      id: 1,
      name: "Travel"
    },
    {
      id: 3,
      name: "Other"
    },
  ];

  const component = renderer.create(
    <Sidebar categories={categories} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});