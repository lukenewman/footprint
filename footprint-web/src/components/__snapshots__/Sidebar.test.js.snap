// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> matches snapshot 1`] = `
<aside
  className="ant-layout-sider ant-layout-sider-dark"
  style={
    Object {
      "flex": "0 0 20%",
      "maxWidth": "20%",
      "minWidth": "20%",
      "width": "20%",
    }
  }
>
  <div
    className="ant-layout-sider-children"
  >
    <ul
      className="ant-menu ant-menu-root ant-menu-inline ant-menu-light"
      data-menu-list={true}
      onKeyDown={[Function]}
      role="menu"
      style={
        Object {
          "borderRight": 0,
          "height": "100%",
        }
      }
      tabIndex={0}
    >
      <li
        className="ant-menu-item ant-menu-item-selected ant-menu-item-only-child"
        data-menu-id={null}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        tabIndex={-1}
      >
        <span
          className="ant-menu-title-content"
        >
          Housing
        </span>
      </li>
      <li
        className="ant-menu-item ant-menu-item-only-child"
        data-menu-id={null}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        tabIndex={-1}
      >
        <span
          className="ant-menu-title-content"
        >
          Travel
        </span>
      </li>
      <li
        className="ant-menu-item ant-menu-item-only-child"
        data-menu-id={null}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        tabIndex={-1}
      >
        <span
          className="ant-menu-title-content"
        >
          Other
        </span>
      </li>
    </ul>
    <div
      aria-hidden={true}
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
</aside>
`;
